scripts/submit_bb.py

```diff
@@ -36,129 +37,178 @@ def gen_command_template(params):
     return command_template_parameters, params.bb
 
 
-def main(args: argparse.Namespace, logger: Logger = get_basic_logger()):
-    params = utils.load_sim_params(os.path.join(args.rel_dir, "sim_params.yaml"))
-    ncores = platform_config[const.PLATFORM_CONFIG.BB_DEFAULT_NCORES.name]
+def main(
+    auto: bool = False,
+    machine: str = host,
+    ncores: int = platform_config[const.PLATFORM_CONFIG.BB_DEFAULT_NCORES.name],
+    rel_dir: Path = Path("."),
```

Comment:
so this is one thing i dont like about python, you cannot really force a argument to be a specific type and raise error instantly if incorrect type is parsed.

So here you are assuming the type to be pathlib.Path, but in the auto_submit.py it is using get_sim_dir, which returns a result of os.path.join ( string type )

Is there a PR for qcore that i'm not aware of? otherwise this seems like it will break, as a many of our code still uses simple string type.

Reply:
In the current PR rel_dir is always passed in as a string. So either the type needs to be changed back to string, the variable cast to Path at the start of the function, or all references to main need to cast to Path before passing the value. This applies to hf, emod3d and post_emod3d too.
Converting sim_struct to use pathlib Paths is a good idea for a future refactor though, we'd have to either test the type or cast to Path at the start of every function though

Reply:
Good point. I looked at auto_submit.py and it relies too much on old os.path. 
Fixing all other references of os.path in auto_submit appears to be beyond the scope.
I'll allow rel_dir and write_directory to be passed as str.